### PR TITLE
updated comments to clarify algorithm

### PR DESCRIPTION
**Pull request by jaurentz**
As per Asgeir's request, I've updated the documentation in `@chebtech/simplify.m` and `@trigtech/simplify.m`.

### CODE REVIEW

File: @chebtech/simplify.m [hunk 1]
```diff
@@ -25,15 +25,19 @@
...
+% STANDARDCHOP also employs a look ahead feature for detecting plateaus. For F
+% with insufficient length the coefficients are padded using prolong. The
+% following parameters are chosen explicitly to work with STANDARDCHOP. 
 % See STANDARDCHOP for details.
 nold = length(f);
 N = max(17, round(nold*1.25 + 5));
 f = prolong(f,N);
 
-% Grab coefficients of F.
+% After the coefficients of F have been padded with zeros an artificial plateau
```
Comment (nickhale) [inline @chebtech/simplify.m]: Is this artificial plateau always required, or only when there has been zero padding of the coefficients?

**Comment by jaurentz** [comment]
It's required whenever there's a zero padding which there always will be.
```matlab
N = max(17,round(1.25*nold + 5));
```

**Comment by nickhale** [comment]
Is there another way to model this noise? 

I realise that in practice, with all the overheads we have elsewhere, it probably makes very little difference, but taking two FFTs here irks me!

**Comment by asgeirbirkis** [comment]
I spotted those two FFTs when profiling related to PR #1625, running the code at the start of issue #1608 . The FFTs are now towards the top of the list in terms of running costs, see attached screenshot of profiler results (lines 3 and 6)
screen shot 2015-10-26 at 13 40 30
.

**Comment by jaurentz** [comment]
We originally started by setting an artificial plateau based on certain parameter values. It became a bit of an engineering problem and ultimately we didn't like how complicated it had become. The goal is for the chopping procedure to simple to implement and explain. We do not claim that the FFT approach is optimal but it definitely satisfies the simplicity requirement. @trefethen and I are currently writing everything up and hopefully we will think up something more clever as we collect our thoughts.

**Comment by jrus** [comment]
For a reader who doesn’t intimately know how the parts fit together, I think this comment is a bit unclear:
```matlab
% After the coefficients of F have been padded with zeros an artificial plateau
% is created using the noisy output from the FFT. The slightly noisy plateau is
% required since STANDARDCHOP uses logarithms to detect plateaus and this has
% undesirable effects when the plateau is made up of all zeros.
```

Zero padding is already a sort of “artificial plateau”, and what are “undesirable effects”?

Maybe something like:
```matlab
% The artificial plateau extension created by padding with zeros is unsuitable for
% STANDARDCHOP which uses logarithms to detect plateaus, and therefore
% breaks when the logarithm is undefined. To avoid pure zeros, perform a forward
% and inverse FFT, relying on the noise added by tiny floating point rounding errors.
```

* * *

What would happen if you padded with e.g. alternating ± `realmin`, instead of all zeros?

**Comment by jrus** [comment]
One slightly related thing, but a bit more general a question: have you folks tried making sure to expand to a power of two when undertaking this kind of operation? In general the FFT is dramatically faster for power-of-two sizes compared to arbitrary sizes. Especially running an FFT on a large prime size is going to result in a considerably more expensive operation.

**Pull request merged**